### PR TITLE
simpson 1/3 rule in c# added

### LINKED ISSUE #1892: Simpson's rule

### PR DESCRIPTION
**Pull request by manildayma**
Fixes #1892 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- 
- 

#### Languages Used:
-c#
- 

#### Files Added:
- 
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Simpsons_rule/simpsons_rule.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
+//simpson's 1/3 rule in C#
```
Comment (jainaman224) [inline Simpsons_rule/simpsons_rule.cs]: Take input from user.

File: Simpsons_rule/simpsons_rule.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        return (float)Math.Log(x); 
+    } 
+  
+    static float simpsons_(float ll, float ul, 
+                                        int n) 
+    { 
+        float h = (ul - ll) / n; 
+  
+        float[] x = new float[10]; 
+        float[] fx= new float[10];
```
Comment (jainaman224) [inline Simpsons_rule/simpsons_rule.cs]: Add a space before `-` as well.

File: Simpsons_rule/simpsons_rule.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        } 
+          
+        res = res * (h / 3); 
+        return res; 
+    } 
+  
+    public static void Main() 
+    {  
+        float lower_limit = 7; 
+        float upper_limit = (float)7.3;
```
Comment (jainaman224) [inline Simpsons_rule/simpsons_rule.cs]: Add comment. Why you have taken this value.

**Comment by manildayma** [comment]
I'll do the necessary changes

**Comment by manildayma** [comment]
fixed by addinf comments and

**Comment by manildayma** [comment]
Fixed!
Added comments; took input from user.

**Comment by manildayma** [comment]
@jainaman224  Please Review this PR.

**Comment by manildayma** [comment]
@jainaman224 @asha15 please review and merge this

**Comment by manildayma** [comment]
@asha15 @jainaman224  please review it. Indentation has been corrected.

File: Simpsons_rule/simpsons_rule.cs [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+		// Calculatiion of x and f(x) 
+		for (int i = 0; i <= n; i++) 
+		{ 
+			x[i] = ll + i * h; 
+			fx[i] = func(x[i]); 
+		} 
+
+		// Result Calculation
+		float res = 0; 
+		for (int i = 0; i <= n; i++)
```
Comment (asha15) [inline Simpsons_rule/simpsons_rule.cs:32]: please keep four space ,don't use tab

File: Simpsons_rule/simpsons_rule.cs [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+/*
+OUTPUT
+Inout lower limit, upper limit and interval : 2
+9
+6
+Result : 11.38688
+*/	
+		
+							
+
```
Comment (asha15) [inline Simpsons_rule/simpsons_rule.cs:74]: remove the extra lines

File: Simpsons_rule/simpsons_rule.cs [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+        Console.WriteLine("Input lower limit, upper limit and internal : ");
+        lower_limit = float.Parse(Console.ReadLine());
+        upper_limit = float.Parse(Console.ReadLine());
+        
+        val = Console.ReadLine();
+        n = Convert.ToInt32(val);
+	Console.WriteLine("Result : ");                             
+	Console.WriteLine(simpsons_(lower_limit, upper_limit, n));
+	Console.WriteLine("Result : ");
```
Comment (asha15) [inline Simpsons_rule/simpsons_rule.cs:61]: please correct the indentation

File: Simpsons_rule/simpsons_rule.cs [hunk 7]
```diff
@@ -0,0 +1,74 @@
+/*simpson's 1/3 rule in C#
+Used for numerical approximation of definite integrals
+*/
+
+using System; 
+
+public class simpsonRule 
+{ 
+
```
Comment (asha15) [inline Simpsons_rule/simpsons_rule.cs:9]: remove this line

**Comment by manildayma** [comment]
Indentation has been corrected as per request. Sorry for the delay.

File: Simpsons_rule/simpsons_rule.cs [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+Used for numerical approximation of definite integrals
+*/
+using System; 
+public class simpsonRule 
+{
+    // Function to calculate f(x) 
+    static float func(float x) 
+    {
+    return (float)Math.Log(x); 
+    }
```
Comment (asha15) [inline Simpsons_rule/simpsons_rule.cs:11]: keep a line space after this

File: Simpsons_rule/simpsons_rule.cs [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+        } 
+    // Result Calculation
+    float res = 0;
+        for (int i = 0; i <= n; i++) 
+        { 
+            if (i == 0 || i == n) 
+	        res += fx[i]; 
+	    else if (i % 2 != 0) 
+	        res += 4 * fx[i]; 
+	    else
```
Comment (asha15) [inline Simpsons_rule/simpsons_rule.cs:32]: please correct the indentation

File: Simpsons_rule/simpsons_rule.cs [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+	        res += 2 * fx[i]; 
+	} 
+	res = res * (h / 3); 
+	return res;
```
Comment (asha15) [inline Simpsons_rule/simpsons_rule.cs:36]: indentation

File: Simpsons_rule/simpsons_rule.cs [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+    { 
+    float h = (ul - ll) / n; 
+    float[] x = new float[10]; 
+    float[] fx = new float[10]; 
+    // Calculatiion of x and f(x) 
+        for (int i = 0; i <= n; i++) 
+        { 
+        x[i] = ll + i * h; 
+        fx[i] = func(x[i]); 
+        }
```
Comment (asha15) [inline Simpsons_rule/simpsons_rule.cs:23]: keep a line space after this

File: Simpsons_rule/simpsons_rule.cs [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+            if (i == 0 || i == n) 
+	        res += fx[i]; 
+	    else if (i % 2 != 0) 
+	        res += 4 * fx[i]; 
+	    else
+	        res += 2 * fx[i]; 
+	} 
+	res = res * (h / 3); 
+	return res; 
+    }
```
Comment (asha15) [inline Simpsons_rule/simpsons_rule.cs:37]: keep a line space after this

File: Simpsons_rule/simpsons_rule.cs [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+    int n;
+    float lower_limit, upper_limit;
+    Console.WriteLine("Input lower limit, upper limit and internal : ");
+    lower_limit = float.Parse(Console.ReadLine());
+    upper_limit = float.Parse(Console.ReadLine());
+    val = Console.ReadLine();
+    n = Convert.ToInt32(val);
+    Console.WriteLine("Result : ");                             
+    Console.WriteLine(simpsons_(lower_limit, upper_limit, n));
+    Console.WriteLine("Result : ");
```
Comment (asha15) [inline Simpsons_rule/simpsons_rule.cs:51]: please correct the indentation

**Comment by manildayma** [comment]
request to review PR. @asha15

**Comment by manildayma** [comment]
Am i supposed to make anymore changes, @asha15  ?

File: Simpsons_rule/simpsons_rule.cs [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+    return (float)Math.Log(x); 
+    } 
+	
+    // Function for approximate integral 
+    static float simpsons_(float ll, float ul, int n) 
+    { 
+    float res = 0;	    
+    float h = (ul - ll) / n; 
+    float[] x = new float[10]; 
+    float[] fx = new float[10];
```
Comment (asha15) [inline Simpsons_rule/simpsons_rule.cs:19]: correct the indentation

File: Simpsons_rule/simpsons_rule.cs [hunk 15]
```diff
@@ -0,0 +1,66 @@
...
+    float res = 0;	    
+    float h = (ul - ll) / n; 
+    float[] x = new float[10]; 
+    float[] fx = new float[10]; 
+	    
+    // Calculatiion of x and f(x) 
+        for (int i = 0; i <= n; i++) 
+        { 
+        x[i] = ll + i * h; 
+        fx[i] = func(x[i]);
```
Comment (asha15) [inline Simpsons_rule/simpsons_rule.cs:25]: indentation

File: Simpsons_rule/simpsons_rule.cs [hunk 16]
```diff
@@ -0,0 +1,66 @@
...
+        { 
+            if (i == 0 || i == n) 
+	        res += fx[i]; 
+	    else if (i % 2 != 0) 
+	        res += 4 * fx[i]; 
+	    else
+	        res += 2 * fx[i]; 
+	} 
+    res = res * (h / 3); 
+    return res;
```
Comment (asha15) [inline Simpsons_rule/simpsons_rule.cs:39]: correct the indentation

File: Simpsons_rule/simpsons_rule.cs [hunk 17]
```diff
@@ -0,0 +1,66 @@
...
+    float lower_limit, upper_limit;
+	    
+    Console.WriteLine("Input lower limit, upper limit and internal : ");
+    lower_limit = float.Parse(Console.ReadLine());
+    upper_limit = float.Parse(Console.ReadLine());
+    val = Console.ReadLine();
+    n = Convert.ToInt32(val);
+    Console.WriteLine("Result : ");                             
+    Console.WriteLine(simpsons_(lower_limit, upper_limit, n));
+    Console.WriteLine("Result : ");
```
Comment (asha15) [inline Simpsons_rule/simpsons_rule.cs:56]: please correct the indentation

**Comment by somya-kapoor** [comment]
Are you working on it?

**Comment by manildayma** [comment]
Yes @somya-kapoor,  got busy in fixing my system

**Comment by manildayma** [comment]
@somya-kapoor  Since I'm not on my system, may I close this PR and create another one with mobile? that won't require any squashing of commits.

File: Simpsons_rule/simpsons_rule.cs [hunk 18]
```diff
@@ -0,0 +1,66 @@
+/*simpson's 1/3 rule in C#
+Used for numerical approximation of definite integrals
+*/
+using System; 
+public class simpsonRule 
+{
+    // Function to calculate f(x) 
+    static float func(float x) 
+    {
+    return (float)Math.Log(x);
```
Comment (somya-kapoor) [inline Simpsons_rule/simpsons_rule.cs:10]: correct the indentation

File: Simpsons_rule/simpsons_rule.cs [hunk 19]
```diff
@@ -0,0 +1,66 @@
...
+    // Function to calculate f(x) 
+    static float func(float x) 
+    {
+    return (float)Math.Log(x); 
+    } 
+	
+    // Function for approximate integral 
+    static float simpsons_(float ll, float ul, int n) 
+    { 
+    float res = 0;
```
Comment (somya-kapoor) [inline Simpsons_rule/simpsons_rule.cs:16]: correct the indentation

File: Simpsons_rule/simpsons_rule.cs [hunk 20]
```diff
@@ -0,0 +1,66 @@
...
+    { 
+    float res = 0;	    
+    float h = (ul - ll) / n; 
+    float[] x = new float[10]; 
+    float[] fx = new float[10]; 
+	    
+    // Calculatiion of x and f(x) 
+        for (int i = 0; i <= n; i++) 
+        { 
+        x[i] = ll + i * h;
```
Comment (somya-kapoor) [inline Simpsons_rule/simpsons_rule.cs:24]: correct the indentation

File: Simpsons_rule/simpsons_rule.cs [hunk 21]
```diff
@@ -0,0 +1,66 @@
...
+        x[i] = ll + i * h; 
+        fx[i] = func(x[i]); 
+        } 
+	    
+    // Result Calculation
+        for (int i = 0; i <= n; i++) 
+        { 
+            if (i == 0 || i == n) 
+	        res += fx[i]; 
+	    else if (i % 2 != 0)
```
Comment (somya-kapoor) [inline Simpsons_rule/simpsons_rule.cs:33]: correct the indentation

File: Simpsons_rule/simpsons_rule.cs [hunk 22]
```diff
@@ -0,0 +1,66 @@
...
+        for (int i = 0; i <= n; i++) 
+        { 
+            if (i == 0 || i == n) 
+	        res += fx[i]; 
+	    else if (i % 2 != 0) 
+	        res += 4 * fx[i]; 
+	    else
+	        res += 2 * fx[i]; 
+	} 
+    res = res * (h / 3);
```
Comment (somya-kapoor) [inline Simpsons_rule/simpsons_rule.cs:38]: correct the indentation

File: Simpsons_rule/simpsons_rule.cs [hunk 23]
```diff
@@ -0,0 +1,66 @@
...
+	        res += 2 * fx[i]; 
+	} 
+    res = res * (h / 3); 
+    return res; 
+    } 
+	
+    // MAIN FUNCTION 
+    public static void Main() 
+    {  
+    string val;
```
Comment (somya-kapoor) [inline Simpsons_rule/simpsons_rule.cs:45]: correct the indentation

**Comment by somya-kapoor** [comment]
Okay

> @somya-kapoor Since I'm not on my system, may I close this PR and create another one with mobile? that won't require any squashing of commits.

Okay

**Pull request closed without merging**